src/main/java/software/amazon/awssdk/s3accessgrants/plugin/S3AccessGrantsPlugin.java

```diff
@@ -168,6 +185,12 @@ public Builder enableFallback(@NotNull Boolean choice) {
            this.enableFallback = choice == null ? DEFAULT_FALLBACK_SETTING: choice;
            return this;
         }
+
+        @Override
+        public Builder userAgent(@NotNull String userAgent) {
+            this.userAgent = userAgent == null ? USER_AGENT : USER_AGENT + "-" + userAgent;
```

Comment:
Do you think we might want to restrict the characters that are allowed while specifying user agent. My concern is that since this is user input we might end up with malicious stuff that I am not sure will be sanitized on the server side or not. Please correct me if this is not a concern. Thanks.

Reply:
You are right. I can add some checks.